chromium_src/chrome/browser/sync/bookmark_sync_service_factory.cc

```diff
@@ -0,0 +1,22 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/sync/bookmark_sync_service_factory.h"
+#include "build/build_config.h"
+
+#if !BUILDFLAG(IS_ANDROID)
+#include "brave/browser/ui/bookmark/bookmark_prefs_service_factory.h"
+
+// Adding BookmarkPrefsServiceFactory as dependency becasue kShowBookmarBar and
+// kAlwaysShowBookmarkBarOnNTP manage bookmar bar state togher and need to
+// register both prefs same time.
+#define DependsOn                                        \
+  DependsOn(BookmarkPrefsServiceFactory::GetInstance()); \
+  DependsOn
+#endif
+#include "src/chrome/browser/sync/bookmark_sync_service_factory.cc"
+#if !BUILDFLAG(IS_ANDROID)
+#undef DependsOn
```

Comment:
what is the reason for this? This is the end of a cc file so why call undef here?

Reply:
just to undef what was defined above, should I remove it?

Reply:
It's unnecessary. This only needs to be done in header files that get included in other files where we don't want the define to apply outside of the file

Reply:
removed here https://github.com/brave/brave-core/pull/15577/commits/c8afa27bf4f2507e62b916dc9dd1fef6d1703927